bigchaindb/backend/mongodb/connection.py

```diff
@@ -113,6 +113,8 @@ def _connect(self):
                                              replicaset=self.replicaset,
                                              serverselectiontimeoutms=self.connection_timeout,
                                              ssl=self.ssl)
+                if self.login is not None and self.password is not None:
```

Comment:
It will still result in an error over at this stage and we will write the same checks that MongoDB driver has.